zebra-state/src/service/check/difficulty.rs

```diff
@@ -314,22 +315,30 @@ impl AdjustedDifficulty {
     /// Zcash specification. (These functions are identical, but they are
     /// specified in slightly different ways.)
     pub fn median_time_past(&self) -> DateTime<Utc> {
-        let median_times: [DateTime<Utc>; POW_MEDIAN_BLOCK_SPAN] = self.relevant_times
-            [0..POW_MEDIAN_BLOCK_SPAN]
-            .try_into()
-            .expect("relevant times is the correct length");
+        let median_times: Vec<DateTime<Utc>> = self
+            .relevant_times
+            .iter()
+            .take(POW_MEDIAN_BLOCK_SPAN)
+            .cloned()
+            .collect();
 
         AdjustedDifficulty::median_time(median_times)
     }
 
     /// Calculate the median of the `median_block_span_times`: the `time`s from a
-    /// slice of `PoWMedianBlockSpan` (11) blocks in the relevant chain.
+    /// Vec of `PoWMedianBlockSpan` (11) or fewer blocks in the relevant chain.
     ///
     /// Implements `MedianTime` from the Zcash specification.
-    pub(crate) fn median_time(
-        mut median_block_span_times: [DateTime<Utc>; POW_MEDIAN_BLOCK_SPAN],
-    ) -> DateTime<Utc> {
+    ///
+    /// # Panics
+    ///
+    /// If provided an empty Vec
+    pub(crate) fn median_time(mut median_block_span_times: Vec<DateTime<Utc>>) -> DateTime<Utc> {
         median_block_span_times.sort_unstable();
-        median_block_span_times[POW_MEDIAN_BLOCK_SPAN / 2]
+
+        // > median(𝑆) := sorted(𝑆)ceiling((length(𝑆)+1)/2)
```

Comment:
Let's put the `ceiling` into a subscript since its output is an index.


```suggestion
        // > median(𝑆) := sorted(𝑆)_{ceiling((length(𝑆)+1)/2)}
```